src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+     *
+     * @param connectionDTO the existing connection Id
+     * @return Boolean
+     */
+    public Boolean testNewConnection(ConnectionDTO connectionDTO)
+        throws BadRequestException, BadGatewayException {
+        DgConnection dgConnection = connectionDTO.getConnection();
+        DgConnection updatedConnection = connectionDTO.getConnection();
+        if (dgConnection.getModule() == null || dgConnection.getModule().toString().equals("")) {
+            throw new BadRequestException(
+                "Connection " + ErrorConstants.ERROR_MODULE_NOT_EMPTY.getErrorMessage(),
+                HttpStatus.BAD_REQUEST);
+        }
+        boolean testConnectionStatus = false;
+        try {
+            if (updatedConnection.getConnectionId() != null) {
+                logger.info("Connection already exists.");
+                /* check if user changed the password or not.*/
+                DgConnection currDbConnection = getConnections(
+                    Optional.of(updatedConnection.getConnectionId()))
+                    .get(0).getConnection();
+                if (currDbConnection.getPassword() != null
+                    && updatedConnection.getPassword() != null) {
+                    if (currDbConnection.getPassword().equals(updatedConnection.getPassword())) {
+                        // set password to "" if the password isn't changed.
+                        // Don't send encrypted password from saas ui to re encrypt.
+                        updatedConnection.setPassword("");
+                    }
+                }
+            }
+            dgConnection.setModule(
+                restDriver.getModuleTypeUsingSourceType(dgConnection.getConnectionType()));
+            dgConnection.setLocation("Cloud");
+            testConnectionStatus = restDriver.testConnection(dgConnection);
+        } catch (DgDrvException e) {
+            logger.error("Test connection failed: " + e);
+            throw new BadGatewayException(
+                ErrorConstants.ERROR_VALIDATING_CONNECTION_DETAILS.getErrorMessage(),
+                HttpStatus.BAD_GATEWAY);
+        }
+        return testConnectionStatus;
+    }
+
+    /**
+     * Fetch metadata of an existing connection by connectionId.
+     *
+     * @param connectionId the existing connection Id
+     * @return List of DatabaseMetadata type.
+     */
+    public List<DatabaseMetadata> fetchExistingConnectionMetadata(Integer connectionId,
+        String moduleType) throws BadRequestException, BadGatewayException {
+        if (connectionId == null) {
+            throw new BadRequestException("Connection id must be specified as path variable.",
+                HttpStatus.BAD_REQUEST);
+        }
+        List<DatabaseMetadata> response = null;
+        try {
+            response = restDriver.getConnectionMetadata(connectionId, moduleType);
+        } catch (DgDrvException e) {
+            logger.error("Error fetching metadata: " + e);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L372
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1184874249>*